test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,168 @@
+	testNamespace = "managed"
+	testNamespaces := []string{testNamespace, "range1", "range2"}
+	defaultTimeoutSeconds = 60
+	By("Create Namespaces if needed")
+	namespaces := clientManaged.CoreV1().Namespaces()
+	for _, ns := range testNamespaces {
+		if _, err := namespaces.Get(context.TODO(), ns, metav1.GetOptions{}); err != nil && errors.IsNotFound(err) {
+			Expect(namespaces.Create(context.TODO(), &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: ns,
+				},
+			}, metav1.CreateOptions{})).NotTo(BeNil())
+		}
+		Expect(namespaces.Get(context.TODO(), ns, metav1.GetOptions{})).NotTo(BeNil())
+	}
+})
+
+func NewKubeClient(url, kubeconfig, context string) kubernetes.Interface {
+	klog.V(5).Infof("Create kubeclient for url %s using kubeconfig path %s\n", url, kubeconfig)
+
+	config, err := LoadConfig(url, kubeconfig, context)
+	if err != nil {
+		panic(err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	return clientset
+}
+
+func NewKubeClientDynamic(url, kubeconfig, context string) dynamic.Interface {
+	klog.V(5).Infof("Create kubeclient dynamic for url %s using kubeconfig path %s\n", url, kubeconfig)
+
+	config, err := LoadConfig(url, kubeconfig, context)
+	if err != nil {
+		panic(err)
+	}
+
+	clientset, err := dynamic.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	return clientset
+}
+
+func LoadConfig(url, kubeconfig, context string) (*rest.Config, error) {
+	if kubeconfig == "" {
+		kubeconfig = os.Getenv("KUBECONFIG")
+	}
+
+	klog.V(5).Infof("Kubeconfig path %s\n", kubeconfig)
+
+	// If we have an explicit indication of where the kubernetes config lives, read that.
+	if kubeconfig != "" {
+		if context == "" {
+			return clientcmd.BuildConfigFromFlags(url, kubeconfig)
+		}
+
+		return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfig},
+			&clientcmd.ConfigOverrides{
+				CurrentContext: context,
+			}).ClientConfig()
+	}
+
+	// If not, try the in-cluster config.
+	if c, err := rest.InClusterConfig(); err == nil {
+		return c, nil
+	}
+
+	// If no in-cluster config, try the default location in the user's home directory.
+	if usr, err := user.Current(); err == nil {
+		klog.V(5).Infof("clientcmd.BuildConfigFromFlags for url %s using %s\n", url,
+			filepath.Join(usr.HomeDir, ".kube", "config"))
+
+		if c, err := clientcmd.BuildConfigFromFlags("", filepath.Join(usr.HomeDir, ".kube", "config")); err == nil {
+			return c, nil
+		}
+	}
+
+	return nil, fmt.Errorf("could not create a valid kubeconfig")
+}
+
+// func deleteCertPolicies(policyNames []string) {
+// 	for _, policyName := range policyNames {
+// 		err := clientManagedDynamic.Resource(gvrCertPolicy).Namespace(testNamespace).Delete(
+// 			context.TODO(), policyName, metav1.DeleteOptions{},
+// 		)
+// 		if !errors.IsNotFound(err) {
+// 			Expect(err).To(BeNil())
+// 		}
+// 	}
+// }
```

Comment:
Should this be commented?

Reply:
I could take it out, but once E2E tests are actually implemented it could be helpful (though it'd also be trivial to copy/paste it from another controller). Okay, I'll take it out. 😆

Reply:
I just figured you commented it while debugging something, and accidentally committed it (maybe I'm projecting my problems here... 😆 )